src/main/java/org/pih/petl/api/JobExecutionTask.java

```diff
@@ -1,47 +1,66 @@
 package org.pih.petl.api;
 
-import org.pih.petl.PetlException;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
 import org.pih.petl.job.PetlJob;
 import org.pih.petl.job.config.JobConfig;
 
+import java.util.Date;
 import java.util.concurrent.Callable;
 
 /**
  * Represents an ETL job execution and the status of this
  */
 public class JobExecutionTask implements Callable<JobExecutionResult> {
 
+    private static final Log log = LogFactory.getLog(EtlService.class);
+
+    private final EtlService etlService;
+    private final PetlJob petlJob;
     private final ExecutionContext executionContext;
     private int attemptNum = 1;
 
-    public JobExecutionTask(ExecutionContext executionContext) {
+    public JobExecutionTask(EtlService etlService, PetlJob petlJob, ExecutionContext executionContext) {
+        this.etlService = etlService;
+        this.petlJob = petlJob;
         this.executionContext = executionContext;
     }
 
     @Override
     public String toString() {
-        return getJobConfig() + " (#" + attemptNum + ")";
+        return executionContext.getJobExecution() + " (#" + attemptNum + ")";
     }
 
     @Override
     public JobExecutionResult call() {
         JobExecutionResult result = new JobExecutionResult(this);
+        JobExecution execution = executionContext.getJobExecution();
+        execution.setStarted(new Date());
+        execution.setStatus(JobExecutionStatus.IN_PROGRESS);
+        etlService.saveJobExecution(execution);
          try {
-             PetlJob job = JobFactory.instantiate(executionContext.getJobConfig());
-             if (job == null) {
-                 throw new PetlException("Unable to find job of type: " + executionContext.getJobConfig().getType());
-             }
-             job.execute(executionContext);
+             log.info(executionContext.getJobExecution());
+             log.info("Job (" + executionContext.getJobExecution().getUuid() + "): " + executionContext.getJobConfig());
+             petlJob.execute(executionContext);
```

Comment:
This is the only place where jobs are actually executed.  So I added tracking of the state here, and also info and error-level logging.  All jobs will have info log done when their execution starts, and error level logging if any exception occurs, even if there are retries, since retrying happens at a higher level than this method.

Reply:
Makes sense to standardize execution into a single location, that had me a little confused before.